api/src/api/userRouter.test.ts

```diff
@@ -343,6 +345,186 @@ describe("userRouter", () => {
         expect(stubTimeStampBusinessSearch.search).toHaveBeenCalledWith(userData.profileData.businessName);
         expect(stubUpdateOperatingPhase).toHaveBeenCalledWith(userData);
       });
+
+      describe("when businessPersona is 'FOREIGN'", () => {
+        it("does not update business name search if dbaBusinessNameAvailability is undefined", async () => {
+          const userData = generateUserData({
+            profileData: generateProfileData({
+              businessPersona: "FOREIGN",
+              needsNexusDbaName: true,
+            }),
+            formationData: generateFormationData({ dbaBusinessNameAvailability: undefined }),
+          });
+          stubUserDataClient.get.mockResolvedValue(userData);
+
+          await request(app).get(`/users/123`).set("Authorization", "Bearer user-123-token");
+          expect(stubTimeStampBusinessSearch.search).not.toHaveBeenCalled();
+        });
+
+        it("does not update dbaBusinessNameAvailability if needsNexusDbaName is false", async () => {
+          const userData = generateUserData({
+            profileData: generateProfileData({
+              businessPersona: "FOREIGN",
+              needsNexusDbaName: false,
+            }),
+            formationData: generateFormationData({
+              dbaBusinessNameAvailability: generateBusinessNameAvailability({
+                status: "AVAILABLE",
+                lastUpdatedTimeStamp: getCurrentDate()
+                  .subtract(1, "hour")
+                  .subtract(1, "minute")
+                  .toISOString(),
+              }),
+            }),
+          });
+          stubUserDataClient.get.mockResolvedValue(userData);
+
+          await request(app).get(`/users/123`).set("Authorization", "Bearer user-123-token");
+          expect(stubTimeStampBusinessSearch.search).not.toHaveBeenCalled();
+        });
+
+        it("updates businessNameAvailability and sets needsNexusDbaName", async () => {
```

Comment:
It would be nice to describe the logical flow here, like "it sets needsNexusDbaName to true when business name becomes unavailable"

Reply:
Agreed. I couldn't connect that phrasing when I wrote this initially.